setup.py

```diff
@@ -68,6 +68,7 @@ def get_version():
         "tiiq": [
             "qblox-instruments==0.7.0",
             "qcodes",
+            "qick",
```

Comment:
Install directly from Git.

Follow example 5. on this page: https://pip.pypa.io/en/stable/cli/pip_install/#examples

Reply:
You can find an in-depth explanation about how to use Git repos with `pip` here https://pip.pypa.io/en/stable/topics/vcs-support/ (in case the example is not explanatory enough).

Reply:
Looks like `qick` is doing something weird during installation:
```
      OSError: [Errno 66] Directory not empty: '/private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/pip-install-6txoxpg_/qick_1e09d75762134716a6c4ac800f0479fc/.eggs/cffi-1.15.1-py3.10-macosx-11.7-x86_64.egg/cffi-1.15.1.dist-info' -> '/private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/pip-install-6txoxpg_/qick_1e09d75762134716a6c4ac800f0479fc/.eggs/cffi-1.15.1-py3.10-macosx-11.7-x86_64.egg/EGG-INFO'
```
https://github.com/qiboteam/qibolab/actions/runs/4489856390/jobs/7896204577?pr=321#step:4:613

I will investigate.

Reply:
#### Explanation
The error above is originated from attempted `cffi` installation to satisfy the `setup_requires` dependencies, i.e. the dependencies declared to be needed while building the package (not to run it, those are `install_requires`).
```py
    setup_requires=["numpy","cffi"],
```
https://github.com/openquantumhardware/qick/blob/aae9fdf5a6d4ae59f8e498a7b6ead6a52c046399/setup.py#L153

GitHub at the moment is complaining that is attempting to write the `cffi` files overwriting a directory that is not empty. That means the `cffi` is somehow already present on the machine, but undetected by `pip` (not sure why...).

This is mostly contained in the error, nothing new, and I still don't know how to solve the issue.

Reply:
According to this issue https://github.com/ampledata/kiss/issues/30 (on a random repo), it seems like the problem would be on Qick side, and they should depend explicitly also on `wheel` (in `setup_requires`).
https://github.com/ampledata/kiss/issues/30

But I'm not so sure that answer is perfectly reliable, nor it applies to our case...

Reply:
https://github.com/pypa/pip/issues/5229

Reply:
Could we try solving this adding ourself the wheel requirement?

Reply:
A more reliable project with a similar problem: https://github.com/pyca/pynacl/issues/453

This really suggests Qick people are missing a `wheel` dep in `setup_requires`.

@rodolfocarobene to debug:
1. fork Qick on your user
2. add the `wheel` dep in `setup_requires`
3. change the Git URL here to your user (instead of `openquantumhardware` org)

Reply:
I think this solves it, checks are still running. Thank you!
I will open an issue on the qick repository, but for now I'm leaving the dependency to the fork.

Reply:
No, it didn't...

Reply:
I saw... Actually without the wheel requirement just 4-5 tests were passing, so I think this is an improvement.
The thing that's weird now is that MacOS build-wheels tests are passing, but not MacOS Tests-qithout QPU.
The only difference I could find in the two tests is [this line](https://github.com/qiboteam/qibolab/actions/runs/4490525190/jobs/7900733569#step:4:22):
```
LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.9.16/x64/lib
```
It's missing in the failing test, but I'm not sure it's the problem

Reply:
Unfortunately, it seems like nothing changed...

> The thing that's weird now is that MacOS build-wheels tests are passing

You're right, they're passing, but it's not weird: they are not attempting to install. The only thing that is done there is to write in a textual list "you will need `qick`", and that's it.

> Actually without the wheel requirement just 4-5 tests were passing, so I think this is an improvement.

I'm sorry, but even this is a chance :face_exhaling: 
In the last run, MacOS was prevented running immediately, because it is a more "precious" resource, we can only use a limited amount of workflows simultaneously on MacOS.
So, while those were waiting, the other ones completed.
Before instead, MacOS workflows were failing immediately, because the installation problem occurs quickly (you do as the first thing), and the other runs were blocked, since there was something already failed.

So, we're still blocked with the same error...

Reply:
Now it's really passing, the trick has been:
```py
    # setup_requires=["numpy","cffi"],
```
https://github.com/AleCandido/qick/blob/2ea4ae6232a81e006bda149bfbfeea19837c3e99/setup.py#L153

Since they have those build deps, but it seems like they are not using anywhere...

I wonder if at this point we should propose a PR (first I should fix the workflow, it has some issues with artifacts update...).

Reply:
Great! Thank you a lot.
I was thinking of opening an issue, but at this point you could also just propose a simple pull request with that commented line couldn't you? In any case, for sure it's not great to have the dependency to a not maintained fork...

Reply:
For sure I don't want to maintain a fork, I will close right after.

Maybe I can actually split the current content in two PRs: one with the comment, and the other with the workflow.
At least I can submit immediately the comment one, and, if ever, we can discuss the workflow separately (it was useful to prove that the wheels could be constructed also on MacOS and Windows, and possibly to have the wheels released at some point, such that we will eventually avoid the git dependency)